app/src/main/java/in/ac/bits_pilani/goa/ard/interfaces/NavigationDrawerListener.java

```diff
@@ -0,0 +1,133 @@
+            final Context context,
+            final TextView navDrawerTitle,
+            final TextView navDrawerSubtitle,
+            final ImageView navDrawerImage,
+            final String tag) {
+        this.context = context;
+        this.navDrawerTitle = navDrawerTitle;
+        this.navDrawerSubtitle = navDrawerSubtitle;
+        this.navDrawerImage = navDrawerImage;
+        this.TAG = tag + " NavigationDrawerListener";
+    }
+
+    @Override
+    public void onDataChange(final DataSnapshot dataSnapshot) {
+        if (dataSnapshot.child(AHC.FDR_NAV_DRAWER_TITLE).exists()) {
+            final String navDrawerTitleText = dataSnapshot
+                    .child(AHC.FDR_NAV_DRAWER_TITLE)
+                    .getValue(String.class);
+
+            if (!Objects.equals(navDrawerTitleText, MainActivity.navDrawerTitleText)) {
+                navDrawerTitle.setText(navDrawerTitleText);
+                MainActivity.navDrawerTitleText = navDrawerTitleText;
+            }
+        }
+        if (dataSnapshot.child(AHC.FDR_NAV_DRAWER_SUBTITLE).exists()) {
+            final String navDrawerSubtitleText = dataSnapshot
+                    .child(AHC.FDR_NAV_DRAWER_SUBTITLE)
+                    .getValue(String.class);
+
+            if (!Objects.equals(navDrawerSubtitleText, MainActivity.navDrawerSubtitleText)) {
+                navDrawerSubtitle.setText(navDrawerSubtitleText);
+                MainActivity.navDrawerSubtitleText = navDrawerSubtitleText;
+            }
+        }
+        if (dataSnapshot.child(AHC.FDR_NAV_DRAWER_IMAGE_LIST).exists()
+                && dataSnapshot.child(AHC.FDR_NAV_DRAWER_IMAGE_LIST).getChildrenCount() > 0) {
+            final ArrayList<String> navDrawerImageList = new ArrayList<>();
+            for (final DataSnapshot childSnapshot
+                    : dataSnapshot.child(AHC.FDR_NAV_DRAWER_IMAGE_LIST).getChildren()) {
+                navDrawerImageList.add(childSnapshot.getValue(String.class));
+            }
+
+            if (!Objects.equals(navDrawerImageList, MainActivity.navDrawerImageList)) {
+                final Random rand = new Random();
+                final String navDrawerImageURL = navDrawerImageList
+                        .get(rand.nextInt(navDrawerImageList.size()));
+                final RequestOptions navDrawerImageOptions = new RequestOptions()
+                        .placeholder(context.getDrawable(R.drawable.nav_drawer_default_image));
+
+                Glide.with(context)
+                        .load(navDrawerImageURL)
+                        .transition(DrawableTransitionOptions.withCrossFade()
+                                .crossFade(MainActivity.NAV_DRAWER_BACKGROUND_ANIM_DUR)
+                        )
+                        .apply(navDrawerImageOptions)
+                        .into(navDrawerImage);
+
+                MainActivity.navDrawerImageURL = navDrawerImageURL;
+                MainActivity.navDrawerImageList = navDrawerImageList;
```

Comment:
do we need to have a navdrawerimagelist in MainActivity? if we can handle all setting from this listener itself?